addons/hr_holidays/models/hr_leave.py

```diff
@@ -208,9 +210,26 @@ def add_follower(self, employee_id):
         if employee.user_id:
             self.message_subscribe_users(user_ids=employee.user_id.ids)
 
+    @api.model
+    def _check_validity(self, values):
+        if 'holiday_status_id' in values:
+            holiday_status = self.env['hr.leave.type'].browse([values['holiday_status_id']])
+
+            if holiday_status.validity_start and holiday_status.validity_stop:
+                vstart = fields.Datetime.from_string(holiday_status.validity_start)
+                vstop  = fields.Datetime.from_string(holiday_status.validity_stop)
+                dfrom  = fields.Datetime.from_string(values.get('date_from', False))
+                dto    = fields.Datetime.from_string(values.get('date_to', False))
+
+                if dfrom and dto and (dfrom < vstart or dto > vstop):
+                    raise UserError('You can take {} only between {} and {}'.format(holiday_status.display_name, \
+                                                                                  holiday_status.validity_start, holiday_status.validity_stop))
+
     @api.model
     def create(self, values):
         """ Override to avoid automatic logging of creation """
+        self._check_validity(values)
```

Comment:
Isn't it possible to make it a real constraint, using `api.constraint` decorator ?

Reply:
I'd need to use something like this @api.constrains('holiday_status_id.valid') which is currently not possible in odoo